apps/backend/src/modules/courses/course.controller.ts

```diff
@@ -132,6 +139,37 @@ class CourseController extends BaseController {
 				params: courseIdParameterValidationSchema,
 			},
 		});
+		this.addRoute({
+			handler: () => {
+				return this.findAll();
+			},
+			method: "GET",
+			path: CoursesApiPath.ALL,
+			preHandler: checkUserPermissions(
+				[PermissionKey.MANAGE_COURSES],
+				PermissionMode.ALL_OF,
+			),
+		});
+		this.addRoute({
+			handler: (options) => {
+				return this.updateByAdmin(
```

Comment:
```suggestion
				return this.update(
```
you need to handle it by permissions